control-plane-operator/hostedclusterconfigoperator/controllers/resources/ingress/reconcile.go

```diff
@@ -164,3 +164,32 @@ func ReconcileDefaultIngressPassthroughRoute(route *routev1.Route, cpService *co
 
 	return nil
 }
+
+// ReconcileDefaultIngressOctaviaService reconciles the default Ingress Service.
+func ReconcileDefaultIngressOctaviaService(ingressService *corev1.Service, ingressFloatingIP string) error {
+	if ingressFloatingIP == "" {
+		return fmt.Errorf("ingress floating IP is empty")
+	}
+
+	ingressService.Spec = corev1.ServiceSpec{}
```

Comment:
This is erasing the exist spec leading to a hot-reconcile loop
```
{"level":"info","ts":"2024-08-05T22:41:07Z","msg":"WARNING: Object got updated more than one time without a no-op update, this indicates hypershift incorrectly reverting defaulted values","type":"*v1.Service","name":"openshift-ingress/octavia-ingress","diff":"  &v1.Service{
  	TypeMeta:   {Kind: "Service", APIVersion: "v1"},
  	ObjectMeta: {Name: "octavia-ingress", Namespace: "openshift-ingress", UID: "47e7e65e-63d9-4291-b2f3-32fec7b5f5dd", ResourceVersion: "1306", ...},
  	Spec: v1.ServiceSpec{
  		Ports: []v1.ServicePort{
  			{
  				... // 3 identical fields
  				Port:       80,
  				TargetPort: {IntVal: 80},
- 				NodePort:   31330,
+ 				NodePort:   0,
  			},
  			{
  				... // 3 identical fields
  				Port:       443,
  				TargetPort: {IntVal: 443},
- 				NodePort:   30416,
+ 				NodePort:   0,
  			},
  		},
  		Selector:  {"ingresscontroller.operator.openshift.io/deployment-ingresscontroller": "default"},
  		ClusterIP: "172.31.230.21",
  		... // 5 identical fields
  		LoadBalancerSourceRanges:      nil,
  		ExternalName:                  "",
- 		ExternalTrafficPolicy:         "Cluster",
+ 		ExternalTrafficPolicy:         "",
  		HealthCheckNodePort:           0,
  		PublishNotReadyAddresses:      false,
  		SessionAffinityConfig:         nil,
  		IPFamilies:                    {"IPv4"},
  		IPFamilyPolicy:                &"SingleStack",
- 		AllocateLoadBalancerNodePorts: &true,
+ 		AllocateLoadBalancerNodePorts: nil,
  		LoadBalancerClass:             nil,
- 		InternalTrafficPolicy:         &"Cluster",
+ 		InternalTrafficPolicy:         nil,
  		TrafficDistribution:           nil,
  	},
  	Status: {},
  }
","semanticDeepEqual":false,"updateCount":10}
```
The spec should be reconciled field by field, not rebuild from scratch.